packages/components/src/components/trace/TraceFilter.vue

```diff
@@ -0,0 +1,354 @@
+
+export default {
+  name: 'v-trace-filter',
+
+  components: {
+    ArrowSearchLeftIcon,
+    ArrowSearchRightIcon,
+    CloseThinIcon,
+    SearchIcon,
+  },
+
+  props: {
+    nodesLength: {
+      type: Number,
+      require: false,
+      default: 0,
+    },
+    currentIndex: {
+      type: Number,
+      default: null,
+    },
+    suggestions: {
+      type: Array,
+      default: () => [],
+    },
+  },
+
+  data() {
+    return {
+      filterValue: '',
+      showSuggestions: false,
+      selectedSuggestion: 0,
+    };
+  },
+
+  watch: {
+    filterValue: {
+      handler(value) {
+        this.$emit('onChange', value);
+      },
+    },
+    selectedSuggestion: {
+      handler(selectedSuggestion) {
+        const selected =
+          this.$refs.suggestionsList.querySelectorAll('li')[selectedSuggestion];
+
+        if (selected.offsetTop < this.$refs.suggestionsList.scrollTop) {
+          this.$refs.suggestionsList.scrollTop = selected.offsetTop;
+        } else if (
+          selected.offsetTop >=
+          this.$refs.suggestionsList.scrollTop +
+            this.$refs.suggestionsList.offsetHeight
+        ) {
+          this.$refs.suggestionsList.scrollTop =
+            selected.offsetTop - selected.offsetHeight * 2;
+        }
+      },
+    },
+  },
+
+  computed: {
+    formClasses() {
+      const classList = ['trace-filter__form'];
+
+      if (this.showSuggestions && this.suggestions.length) {
+        classList.push('trace-filter__form--show-suggestions');
+      }
+
+      return classList;
+    },
+  },
+
+  methods: {
+    setValue(newValue) {
+      this.filterValue = newValue;
+    },
+    clearValue() {
+      this.filterValue = '';
+    },
+    onPrevArrow() {
+      this.$emit('onPrevArrow');
+    },
+    onNextArrow() {
+      this.$emit('onNextArrow');
+    },
+    getSuggestionClasses(index) {
+      const classes = ['trace-filter__suggestions-item'];
+      if (index === this.selectedSuggestion) {
+        classes.push('trace-filter__suggestions-item--selected');
+      }
+      return classes;
+    },
+    onInputFocus() {
+      this.showSuggestions = true;
+      this.selectedSuggestion = 0;
+    },
+    onFormSubmit() {
+      if (
+        this.$refs.input === window.document.activeElement &&
+        this.showSuggestions &&
+        this.suggestions.length
+      ) {
+        this.makeSelection(this.suggestions[this.selectedSuggestion].id);
+      }
+    },
+    makeSelection(eventId) {
+      this.filterValue = `${this.filterValue} id:${eventId}`.trim();
+      this.showSuggestions = false;
+    },
+    onWindowClick(event) {
+      if (!getEventTarget(event.target, this.$refs.form)) {
+        this.showSuggestions = false;
+      }
+    },
+    onWindowKeyUp(event) {
+      if (this.$refs.input !== window.document.activeElement) {
+        return;
+      }
+
+      switch (event.key) {
+        case 'Escape':
+          this.showSuggestions = false;
```

Comment:
We should clear the focus on the input as well. Otherwise the user needs to drop and regain focus to bring the suggestion list back.